Complete Mad Libs assignment

RyanSpittler opened the pull request:
[#88836354]

RyanSpittler committed:
Complete Mad Libs assignment

[#88836354]

RyanSpittler committed:
Replace .chomp with .strip

[#88836354]

jasonnoble commented on methods_and_recursion/mad_libs.rb:
```diff
@@ -0,0 +1,70 @@
+
+# This helps to check whether 'a' or 'an' will be used.
+class String
+  def vowel?
```
Please don't monkey patch for these assignments...

jasonnoble commented on methods_and_recursion/mad_libs.rb:
```diff
@@ -0,0 +1,70 @@
+class String
+  def vowel?
+    vowels = %w(a e i o u A E I O U)
+    if vowels.include?(self)
```
Pro Tip:

```
if ________
  true
else
  false
end
```

can be rewritten as:

```
___________
```

jasonnoble commented on methods_and_recursion/mad_libs.rb:
```diff
@@ -0,0 +1,70 @@
+  end
+end
+
+def ask_for_word(term, list_of_words)
```
How about an optional third parameter `seen_before` which defaults to false?

```
another_word = seen_before ? 'a' : 'another'
```

jasonnoble commented on methods_and_recursion/mad_libs.rb:
```diff
@@ -0,0 +1,70 @@
+                ]
+
+needed_blanks.each do |term|
+  mad_libs << ask_for_word(term, already_requested)
```
Won't `already_requested` always be the empty array from line 27?  You need to add term to `already_requested` somewhere.

RyanSpittler replied:
It was hidden on line 21 of the `ask_for_words` method.

jasonnoble commented on methods_and_recursion/mad_libs.rb:
```diff
@@ -0,0 +1,70 @@
+  mad_libs << ask_for_word(term, already_requested)
+end
+
+puts "Here is your madlib:"
```
You can DRY up your code a bit:

```
puts <<END_OF_MADLIB
  A vacation is when you take a trip to some #{mad_libs[0] place
  with your #{mad_libs[1]} family. Usually you go to some place
  ...
END_OF_MADLIB
```

jasonnoble commented:
Rebase me...

RyanSpittler committed:
Remove class defined methods

- Removed method to check whether a string is a vowel or not.
- Dryed up code.

[#88836354]
